backend/service/temporal/temporal.go

```diff
@@ -0,0 +1,81 @@
+package temporal
+
+// <!-- START clutchdoc -->
+// description: Workflow client for temporal.io.
+// <!-- END clutchdoc -->
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/uber-go/tally"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/log"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	temporalv1 "github.com/lyft/clutch/backend/api/config/service/temporal/v1"
+	"github.com/lyft/clutch/backend/service"
+)
+
+const Name = "clutch.service.temporal"
+
+func New(cfg *anypb.Any, logger *zap.Logger, scope tally.Scope) (service.Service, error) {
+	config := &temporalv1.Config{}
+	if err := cfg.UnmarshalTo(config); err != nil {
+		return nil, err
+	}
+	return newClient(config, logger, scope)
+}
+
+type ClientManager interface {
+	GetNamespaceClient(namespace string) (client.Client, error)
+}
+
+func newClient(cfg *temporalv1.Config, logger *zap.Logger, scope tally.Scope) (ClientManager, error) {
+	ret := &clientImpl{
+		hostPort:       fmt.Sprintf("%s:%d", cfg.Host, cfg.Port),
+		metricsHandler: newMetricsHandler(scope),
+		logger:         newTemporalLogger(logger),
+
+		// Explicitly disable the healthcheck (i.e. connect lazily) so service instantiation will not block server startup.
+		copts: client.ConnectionOptions{DisableHealthCheck: true},
```

Comment:
in an application that only talks to temporal, it makes sense. so i guess if you were running a clutch instance with only a temporal worker it might be useful to crash the application when temporal is down. i will add it.